Fix `qml.matrix` with multiple broadcasted operations in a tape (#3025)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Improvements</h3>

* `qml.matrix` now can also compute the matrix of tapes/QNodes that contain multiple

>   broadcasted operations, as they are present in decompositions of broadcasted operations.

Reviewer: > as they are present in decompositions of broadcasted operations.

This reads a bit confusing, what does this refer to?

Author: Ah, agreed :D It refers to this problem/application popping up when we do
```python
with qml.tape.QuantumTape() as tape:
    qml.IsingXX(np.array([0.4, 2.1, -0.4]), wires=[1, 2]).decomposition()
qml.matrix(tape)
```
i.e. when we decompose a broadcasted operation and then want to compute the matrix of the resulting tape.

Author: Tried to make it more explicit/clear

Reviewer: Thanks! Left a suggestion.

<h3>Improvements</h3>

* `qml.matrix` now can also compute the matrix of tapes/QNodes that contain multiple

>   broadcasted operations, as they are present in decompositions of broadcasted operations.

Reviewer: ```suggestion
  broadcasted operations and/or the decompositions of broadcasted operations.
```

Author: I had already rewritten it a bit, to be more explicit in what I meant. Does this work now?

<h3>Improvements</h3>

* `qml.matrix` now can also compute the matrix of tapes/QNodes that contain multiple
  broadcasted operations, or non-broadcasted operations after broadcasted ones.
  [(#3025)](https://github.com/PennyLaneAI/pennylane/pull/3025)

>   A common scenario in which this becomes relevant is the decomposition of broadcasted

Reviewer: Nice! :) This reads very well.

>   operations: The decomposition in general will contain one or multiple broadcasted

Reviewer: ```suggestion
  operations: the decomposition in general will contain one or multiple broadcasted
```
